### PR TITLE
Geometry Primitives Revamp

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/5_Plat/Geo_Pri.mdx [hunk 1]
```diff
@@ -1,14 +1,213 @@
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// BeginCodeSnip{Point Class}
+struct Point {
+	int x, y;
+	Point(int a = 0, int b = 0) : x(a), y(b) {}
+
+	friend istream &operator>>(istream &in, Point &p) {
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:47]: is `friend` needed? all the members are public anyways

**Comment by SansPapyrus683** [comment]
i did push some minor changes, code is pretty nice

File: content/5_Plat/Geo_Pri.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:47]: It gives an error without friend

Comment (SansPapyrus683) [reply content/5_Plat/Geo_Pri.mdx:47]: i think that's cause you have to move it out of the class itself, but whatever

Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:47]: Yep, this way works. Do you want it outside or inside?

Comment (SansPapyrus683) [reply content/5_Plat/Geo_Pri.mdx:47]: now that i think about it it doesn't matter, keep it as it is

File: content/5_Plat/Geo_Pri.mdx [hunk 2]
```diff
@@ -1,14 +1,213 @@
...
+Let $m_1=\frac{P_1.y - P_2.y}{P_1.x - P_2.x}$ be the slope of the linear function that goes through the points $P_1$ and $P_2$ and
+$m_2=\frac{P_2.y - P.y}{P_2.x - P.x}$ be the slope of the linear function that goes through the points $P_2$ and $P$.
+
+The points are collinear if and only if the slopes are equal i.e. $m_1 = m_2 \Leftrightarrow \frac{P_1.y - P_2.y}{P_1.x - P_2.x} = \frac{P_2.y - P.y}{P_2.x - P.x}
+\Leftrightarrow (P_1.y - P_2.y) * (P_2.x - P.x) = (P_1.x - P_2.x) * (P_1.x - P_2.x) \ Leftrightarrow (P_1.y - P_2.y) * (P_2.x - P.x) - (P_1.x - P_2.x) * (P_1.x - P_2.x) = 0$.
+And this way we obtained the previously mentioned formula.
+
+This formula doesn't only tell us whether the points are collinear, but where the point $P$ is located too.
+
+![Dem](/assets/dem-collinear.png)
```
Comment (ryanchou-dev) [inline content/5_Plat/Geo_Pri.mdx:29]: love this drawing

File: content/5_Plat/Geo_Pri.mdx [hunk 3]
```diff
@@ -1,14 +1,213 @@
...
+</CPPSection>
+</LanguageSection>
+
+<FocusProblem problem="polygon" />
+
+The following algorithm uses the [Shoelace formula](https://en.wikipedia.org/wiki/Shoelace_formula).
+
+<Spoiler title="Demonstration">
+To Do
+</Spoiler>
```
Comment (ryanchou-dev) [inline content/5_Plat/Geo_Pri.mdx:168]: do you plan to finish this in this pr or in a different one?

**Review by ryanchou-dev — COMMENTED** [review]
great! why do you have two uploads of the image (one in src, one in the division)?

File: content/5_Plat/Geo_Pri.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:29]: Thank you :))

File: content/5_Plat/Geo_Pri.mdx (hunk 3 above)
Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:168]: Finish in this pr

File: content/5_Plat/Geo_Pri.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:29]: I removed the image in src.

File: content/5_Plat/Geo_Pri.mdx [hunk 4]
```diff
@@ -1,14 +1,320 @@
...
+### Implementation
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int INF = 2e9;
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:225]: can you use `INT32_MAX` instead

Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx]: Done

**Comment by SansPapyrus683** [comment]
ok i just need screenshots of AC with the _current_ solutions and i'll merge

**Comment by brebenelmihnea** [comment]
image
image
image

**Comment by brebenelmihnea** [comment]
image

**Comment by SansPapyrus683** [comment]
and this is the new code?

**Comment by brebenelmihnea** [comment]
Yes

**Pull request merged**